### DISCUSSION TITLE
How to enable social media logins in self hosted version?

### DISCUSSION THREAD

**Discussion by gauravkaliadev** [supabase/supabase#11855] [Questions] [answered]
Hi All,

I have been integrating Google and FB login in my app with the self-hosted version.

I have tried to update the configurations using the studio but updates are not allowed in the self-hosted version.

I tried to find the solution to this using the env variable but no luck so far.

If I am missing something please let me know, any help is appreciated.

Thanks
Gaurav

**Comment by gauravkaliadev** [comment, marked as answer]
Correct configurations to enable google auth provider is below which need to be added to the docker-compose file.

 ```
      GOTRUE_EXTERNAL_GOOGLE_ENABLED: ${ENABLE_GOOGLE_SIGNUP}
      GOTRUE_EXTERNAL_GOOGLE_CLIENT_ID: ${GOOGLE_CLIENT_ID}
      GOTRUE_EXTERNAL_GOOGLE_SECRET: ${GOOGLE_CLIENT_SECRET}
      GOTRUE_EXTERNAL_GOOGLE_REDIRECT_URI: ${GOOGLE_REDIRECT_URI}
```

I hope this will help someone.

Thanks

**Reply by brunolm** [reply]
Where should I put this? Doesn't work under supabase-auth

**Reply by brunolm** [reply]
So, it works, but the UI doesn't show it works.

**Reply by gauravkaliadev** [reply]
After you add above configs to docker compose, make sure you are adding the values in .env file for all the parameters.

These changes will not reflect on supabase studio UI.

If the configurations are correct, you should see the gmail auth screen.

Apologies for replying late and I hope it's working for you already now.
